### PR TITLE
Steven new test utilities

### PR DESCRIPTION
**Pull request by smarquez-jpg**

### CODE REVIEW

**Review by tayyaba17 — COMMENTED** [review]
LGTM

File: portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  /*@Test
+  public void updateTagsToSearchTest() {
+    List<String> tagsToSearch = new ArrayList<String>();
+    TagsUtils.updateTagsToSearch(tagsToSearch);
+    //TODO find out what email is used to get tags
+    UserService userService = UserServiceFactory.getUserService();
+    String email = (String) userService.getCurrentUser().getEmail();
+    System.out.println(email);
+    Assert.assertEquals(1, 0);
+  }*/
```
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java]: Remove?

File: portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    blogMessageEntity.setProperty(BlogConstants.TIME, System.currentTimeMillis());
+    blogMessageEntity.setProperty(TAG, tag);
+    blogMessageEntity.setProperty(BlogConstants.PARENTID_PARAMETER, parentID);
+    return blogMessageEntity;
+  }
+
+  @Test
+  public void createDatastoreEntitiesWithResponse() {
+    BlogMessage parent = new BlogMessage (1, "#business", "test text business", "nickname", HAS_TAGS_EMAIL, new ArrayList<BlogMessage>(), 0, 0);
+    BlogMessage child = new BlogMessage (2, "#business", "test text business", "nickname", HAS_TAGS_EMAIL, new ArrayList<BlogMessage>(), 0, parent.getMessageId());
```
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java]: This line is probably too long?

**Review by peizhaoou — APPROVED** [review]
LGTM overall

File: portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    Assert.assertEquals(2, actual.size());
+  }
+
+  @Test
+  public void loadAllBlogsOrLastTestFalse() {
+    createDatastoreEntities();
+    List<BlogMessage> actual = DatastoreUtils.LoadAllBlogsOrLast(false);
+
+    Assert.assertEquals(1, actual.size());
```
Comment (martiKallas) [inline portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java]: You should check that this is the message you expect.

File: portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  @Test
+  public void updateTagsToSearchTest() {
+    List<String> tagsToSearch = new ArrayList<String>();
+    createDatastoreEntities();
+    TagsUtils.updateTagsToSearch(tagsToSearch);
+   
+    Assert.assertEquals(3, tagsToSearch.size());
```
Comment (martiKallas) [inline portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java]: Here too, make sure this contains the tags you expect.

**Review by martiKallas — APPROVED** [review]

**Pull request merged**